src/ingest/SpecRepo.cpp

```diff
@@ -137,6 +139,26 @@ void genSpecs4Swap(const std::string& version,
   LOG(WARNING) << "only s3 supported for now";
 }
 
+inline void genSpecs4HourlyRoll(const std::string& version,
+                           const TableSpecPtr& table,
+                           const size_t endHour, const size_t currDay,
+                           std::vector<std::shared_ptr<IngestSpec>>& specs) {
+  const auto now = Evidence::now();
+  // parse location to get protocol, domain/bucket, path
+  auto sourceInfo = nebula::storage::parse(table->location);
+  auto fs = nebula::storage::makeFS("s3", sourceInfo.host);
+
+  // scan hourly partition of last day
+  for(size_t j = 0 ; j < endHour ; j++) {
+    auto timeValue = now - currDay * DAY_SECONDS;
```

Comment:
update the timeValue to reflect the hour value.

(Note, this is only useful when use macro value as time value, if time value is coming from a column, this value is useless)

Reply:
actually, let's just keep using timeValue as timestamp for now. We might looks into support timestamp extraction from column if needed.